modules/java/generator/gen_java.py

```diff
@@ -182,8 +181,20 @@ def libVersionBlock():
 """
 
 class GeneralInfo():
-    def __init__(self, name, namespaces):
-        self.namespace, self.classpath, self.classname, self.name = self.parseName(name, namespaces)
+    def __init__(self, type, decl, namespaces):
+        self.namespace, self.classpath, self.classname, self.name = self.parseName(decl[0], namespaces)
+
+        # parse doxygen comments
+        self.params={}
+        if type == "class":
+            docstring="// C++: class " + self.name + "\n//javadoc: " + self.name
+        else:
+            docstring=""
+        if len(decl)>5 and decl[5]:
+            if re.search("(@|\\\\)deprecated", decl[5]):
+                docstring += "\n@Deprecated"
```

Comment:
It is better to create additional `annotation` field instead of putting it to the `docstring`.